Source/SwiftyCamViewController.swift

```diff
@@ -363,712 +421,679 @@ open class SwiftyCamViewController: UIViewController {
+      super.viewDidAppear(animated)
 
-		// Subscribe to device rotation notifications
+        // Subscribe to device rotation notifications
 
-		if shouldUseDeviceOrientation {
-			orientation.start()
-		}
+        if shouldUseDeviceOrientation {
+            orientation.start()
+        }
 
-		// Set background audio preference
+        // Set background audio preference
 
-		setBackgroundAudioPreference()
+        setBackgroundAudioPreference()
 
-		sessionQueue.async {
-			switch self.setupResult {
-			case .success:
-				// Begin Session
-				self.session.startRunning()
-				self.isSessionRunning = self.session.isRunning
+        sessionQueue.async {
+            switch self.setupResult {
+            case .success:
+                // Begin Session
+                self.session.startRunning()
+                self.isSessionRunning = self.session.isRunning
 
                 // Preview layer video orientation can be set only after the connection is created
                 DispatchQueue.main.async {
                     self.previewLayer.videoPreviewLayer.connection?.videoOrientation = self.orientation.getPreviewLayerOrientation()
                 }
 
-			case .notAuthorized:
+            case .notAuthorized:
+                // Prompt to App Settings
                 if self.shouldPrompToAppSettings == true {
                     self.promptToAppSettings()
                 } else {
                     self.cameraDelegate?.swiftyCamNotAuthorized(self)
                 }
-			case .configurationFailed:
-				// Unknown Error
-                DispatchQueue.main.async {
-                    self.cameraDelegate?.swiftyCamDidFailToConfigure(self)
-                }
-			}
-		}
-	}
+            case .configurationFailed:
+                // Unknown Error
+                DispatchQueue.main.async(execute: { [unowned self] in
+                    let message = NSLocalizedString("Unable to capture media", comment: "Alert message when something goes wrong during capture session configuration")
+                    let alertController = UIAlertController(title: "AVCam", message: message, preferredStyle: .alert)
+                    alertController.addAction(UIAlertAction(title: NSLocalizedString("OK", comment: "Alert OK button"), style: .cancel, handler: nil))
+                    self.present(alertController, animated: true, completion: nil)
+                })
+            }
+        }
+    }
 
-	// MARK: ViewDidDisappear
+    // MARK: ViewDidDisappear
 
-	/// ViewDidDisappear(_ animated:) Implementation
+    /// ViewDidDisappear(_ animated:) Implementation
 
 
-	override open func viewDidDisappear(_ animated: Bool) {
-		super.viewDidDisappear(animated)
+    override open func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
 
-        NotificationCenter.default.removeObserver(self)
-        sessionRunning = false
+        // If session is running, stop the session
+//        if self.isSessionRunning == true {
```

Comment:
I think we still want the session to stop running when the vc disappears

Reply:
Ah, yes you're right. This edit may have been too coupled to my use case.